src/analysisd/alerts/exec.c

```diff
@@ -157,14 +219,156 @@ void OS_Exec(int execq, int *arq, const Eventinfo *lf, const active_response *ar
                     merror("Error communicating with ar queue (%d).", rc);
                 }
             }
-        }
+        } 
     }
 
     cleanup:
 
     /* Clean up Memory */
-    free(filename);
-    free(extra_args);
+    os_free(filename);
+    os_free(extra_args);
 
     return;
 }
+
+/**
+ * @brief extract word from a string between two words
+ *
+ * @param[in] sentence The string where the words are found.
+ * @param[in] word1 Word on the left side of the word to extract.
+ * @param[in] word2 Word on the right side of the word to extract.
+ * @return The word to extract or NULL on failure, remember to free memory after return
+ */
+char *extract_word_between_two_words(const char* sentence, const char* word1, const char* word2)
+{
+    const char *p1 = strstr(sentence, word1);
+    if(!p1) {
+        return NULL;
+    }
+    p1++;
```

Comment:
What if the word has more than one character? I think you should increment the pointer the size of the `word1`.